File: design-pattern-go/src/patterns/creational/abstract_factory/ToyotaFactory.go
package abstract_factory

type ToyotaFactory struct {
	VehicleFactory
}

func (f *ToyotaFactory) CreateDeportiveCar() DeportiveCar {
	return ToyotaYaris{}
}
func (f *ToyotaFactory) CreateRusticCar() RusticCar {
	return ToyotaForRunner{}
}

File: design-pattern-ts/src/patterns/creator/abstract-factory/ToyotaFactory.ts
import { DeportCar } from "./DeportCar"
import { RusticCar } from "./RusticCar"
import { VehicleFactory } from "./VehicleFactory"
import { ToyotaYaris } from "./ToyotaYaris"
import { ToyotaForRunner } from "./ToyotaForRunner"

export class ToyotaFactory implements VehicleFactory {
	produceDeportCar(): DeportCar {
		return new ToyotaYaris()
	}
	produceRusticCar(): RusticCar {
		return new ToyotaForRunner()
	}
}
